libs/blocks/caas/utils.js

```diff
@@ -208,15 +208,43 @@ const getFilterObj = ({ excludeTags, filterTag, icon, openedOnLoad }, tags, stat
   return filterObj;
 };
 
+const getCustomFilterObj = ({ group, filtersCustomItems, openedOnLoad }) => {
+  if (!group) return null;
+
+  const items = filtersCustomItems.map((item) => ({
+    id: item.customFilterTag[0],
+    label: item.filtersCustomLabel,
+  }));
+
+  const filterObj = {
+    id: group,
+    openedOnLoad: !!openedOnLoad,
+    items,
+    group,
+  };
+
+  return filterObj;
+};
+
 const getFilterArray = async (state) => {
-  if (!state.showFilters || state.filters.length === 0) {
+  if ((!state.showFilters || state.filters.length === 0) && state.filtersCustom.length === 0) {
     return [];
   }
 
   const { tags } = await getTags(state.tagsUrl);
-  const filters = state.filters
-    .map((filter) => getFilterObj(filter, tags, state))
-    .filter((filter) => filter !== null);
+  const useCustomFilters = state.filterBuildPanel === 'custom';
+
+  let filters = [];
+  if (!useCustomFilters) {
+    filters = state.filters
+      .map((filter) => getFilterObj(filter, tags, state))
+      .filter((filter) => filter !== null);
+  } else {
+    filters = state.filtersCustom.length > 0
+      ? state.filtersCustom.map((filter) => getCustomFilterObj(filter, state))
```

Comment:
You only destructure the first argument on `getCustomFilterObj` do you need to pass state?

Reply:
Good catch! :)